### PR TITLE
Frankie/core#v0.3.0.rc 1

### PR DESCRIPTION
**Pull request by frankiebee**
This PR updates to the latest version of core (still in rc)

### CODE REVIEW

File: src/signing/index.ts [hunk 1]
```diff
@@ -327,25 +308,17 @@ export default class SignatureRequestManager {
...
+    // Extract the required fields from parsedMsg
+    const payload = JSON.parse(message.msg)
+    console.log('EncMsg', message)
+    console.log('payload', payload)
+    const sigProof = (await sendHttpPost(
+      `http://${message.url}/user/relay_tx`,
+      JSON.stringify(payload)
+    ))
+    console.log('fetch returned:',sigProof, new Date(Date.now()))
+    return sigProof
```
[the remark this answers is not in the document]
Comment (mixmix) [reply src/signing/index.ts:321]: If you wanna introspect I like

```js
.then(res => {
  console.log('stuff')
  return res
})
```

File: src/signing/index.ts [hunk 2]
```diff
@@ -355,59 +328,32 @@ export default class SignatureRequestManager {
...
-      }
-    )
-
-    return validatorsInfo
+  async pickValidator (validators: string[], signingGroup: string[]): Promise<ValidatorInfo> {
+    const relayers = validators.reduce((agg, stashKey) => {
+      if (signingGroup.includes(stashKey)) return agg
+      agg.push(stashKey)
+      return agg
+    }, [])
```
[the remark this answers is not in the document]
Comment (mixmix) [reply src/signing/index.ts:336]: could even make it `pickRelayer(validators, signers)`

File: tests/four-nodes.test.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    keyring,
+    endpoint: 'ws://127.0.0.1:9944',
+  })
+
+  await run('entropy ready', entropy.ready)
+  await run('jump Start Network', jumpStartNetwork(entropy))
+  const validators = (await run('validators', entropy.substrate.query.session.validators())).toHuman()
+  const signingGroup = (await run('signingGroup', entropy.substrate.query.stakingExtension.signers())).toHuman()
+  console.log('validators, signingGroup', validators, signingGroup)
+  t.equal(validators.length, 4, 'expecting 4 validators in validator set')
```
Comment (mixmix) [inline tests/four-nodes.test.ts:55]: We can do better than this I think. If you sort that list it should be deterministic right?

**Review by mixmix — CHANGES_REQUESTED** [review]
This Review:
- :fire: couple gotchas / need change
- :heavy_check_mark: some `console.log` scrubbing
- :warning: a TODO to remember for when `-rc1` is removed
- some style / requests (casual)

File: tests/testing-utils/network.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  try {
+    execFileSync('dev/bin/spin-down.sh', [networkType], {
+      shell: true,
+      cwd: process.cwd(),
+      stdio: 'inherit',
+    })
+  } catch (e) {
+    console.error('Error in afterAll: ', e.message)
+  }
+}
```
[the remark this answers is not in the document]
Comment (frankiebee) [reply tests/testing-utils/network.ts:26]: this snuck in from the old branc removing it

File: package.json [hunk 5]
```diff
@@ -107,7 +107,7 @@
   "dependencies": {
     "@entropyxyz/entropy-protocol-nodejs": "^0.2.0",
     "@entropyxyz/entropy-protocol-web": "^0.2.0",
-    "@polkadot/api": "^10.13.1",
+    "@polkadot/api": "^14.0.1",
```
[the remark this answers is not in the document]
Comment (frankiebee) [reply package.json:110]: yes this was done on purpose. but we are good with it

File: src/signing/index.ts [hunk 6]
```diff
@@ -327,25 +306,14 @@ export default class SignatureRequestManager {
...
-        sigProof.push(message.tss_account)
-        return sigProof
-      })
-    )
+  async submitTransactionRequest (message: EncMsg): Promise<string[][]> {
+    // Extract the required fields from parsedMsg
+    const payload = JSON.parse(message.msg)
+    const sigProof = (await sendHttpPost(
+      `http://${message.url}/user/relay_tx`,
+      JSON.stringify(payload)
```
Comment (mixmix) [inline src/signing/index.ts:314]: Why do you parse + stringify?
It doesn't make a difference, ... unless you hit a Uint8Array in which case it makes it into an ugly ?

**Comment by mixmix** [comment]
Testing error:

```bash
     ✔ network up ------------------------------ 31.12s
    ✔ wasm ------------------------------------ 0s
    ✔ admin account should have an address and for now it should match registrations address
    ✔ got right address
    ✔ entropy ready --------------------------- 0.232s

    ✖ jump-start network
    ---------------------
      operator: error
      at: <anonymous> (/home/runner/work/sdk/sdk/tests/testing-utils/index.ts:64:11)
      stack: |-
  Error: jump-start network timed out
  at file:///home/runner/work/sdk/sdk/dev/testing-utils.mjs:82:43
  at new Promise (<anonymous>)
  at jumpStartNetwork (file:///home/runner/work/sdk/sdk/dev/testing-utils.mjs:81:18)
  at Test.<anonymous> (/home/runner/work/sdk/sdk/tests/end-to-end.test.ts:56:35)
  at process.processTicksAndRejections (node:internal/process/task_queues:95:5)


    ✖ register
    -----------
      operator: error
      at: <anonymous> (/home/runner/work/sdk/sdk/tests/testing-utils/index.ts:64:11)
      stack: |-
  Error: registry.JumpStartNotCompleted:
  at <anonymous> (/home/runner/work/sdk/sdk/src/extrinsic/index.ts:59:22)
  at file:///home/runner/work/sdk/sdk/node_modules/@polkadot/api/promise/decorateMethod.js:56:28
  at file:///home/runner/work/sdk/sdk/node_modules/@polkadot/util/nextTick.js:13:13
  ```

File: dev/docker-scripts/four-nodes.yaml [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+
+services:
+  alice-tss-server:
+    # This pulls from Docker Hub,  so you can replace the image tag name
+    # (the string after the `:` in the `image` key) with chatCharlier tag
+    # name you'd like to use. Each Docker image is built from a specific
+    # commit in the `entropy-core` repository. Refer to the wiki page at
+    # https://github.com/entropyxyz/meta/wiki/Git#conventions for details
+    # about how Git tags, branches, and commits refer to image tags here.
+    image: entropyxyz/entropy-tss:${ENTROPY_CORE_VERSION:-latest}
```
[the remark this answers is not in the document]
Comment (frankiebee) [reply dev/docker-scripts/four-nodes.yaml:20]: this `-latest` came as a suggestion from @vitropy i believe i am hestant to remove this as i want to get the latest build for this version tag that i am asking for

File: tests/end-to-end.test.ts [hunk 8]
```diff
@@ -7,23 +7,24 @@ import * as util from '@polkadot/util'
...
 const msg = Buffer.from('Hello world: signature from entropy!').toString('hex')
 
-test('End To End', async (t) => {
+test.only('End To End', async (t) => {
   const run = promiseRunner(t)
   await run('network up', spinNetworkUp(networkType))
   t.teardown(async () => {
     await entropy.close()
-    await spinNetworkDown(networkType).catch((error) =>
+    if (!process.env.DONT_KILL) await spinNetworkDown(networkType).catch((error) =>
```
[the remark this answers is not in the document]
Comment (frankiebee) [reply tests/end-to-end.test.ts:27]: done also env is now `ENTROPY_DONT_KILL`

File: tests/four-nodes.test.ts (hunk 3 above)
Comment (frankiebee) [reply tests/four-nodes.test.ts:53]: but if the validator list changes then this is a hardcoded value that needs to change.

File: tests/four-nodes.test.ts [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+  const entropy = new Entropy({
+    keyring,
+    endpoint: 'ws://127.0.0.1:9944',
+  })
+
+  await run('entropy ready', entropy.ready)
+  await run('jump Start Network', jumpStartNetwork(entropy))
+  const validators = (await run('validators', entropy.substrate.query.session.validators())).toHuman()
+  const signingGroup = (await run('signingGroup', entropy.substrate.query.stakingExtension.signers())).toHuman()
```
[the remark this answers is not in the document]
Comment (frankiebee) [reply tests/four-nodes.test.ts:52]: i dont see it as a "mess" worth fixing. the point of this test is to ensure that the docker script can run and that jump start works. if i was using it in more places i would consider this. but the point of the `.substrate` is to expose the api of polkadotjs or whatever underlying wrapper for the network communication.

File: src/signing/index.ts (hunk 6 above)
Comment (frankiebee) [reply src/signing/index.ts]: fixed thanks @mixmix for the catch 😬

**Review by mixmix — DISMISSED** [review]

File: package.json [hunk 10]
```diff
@@ -64,7 +64,8 @@
     "postinstall": "patch-package",
     "prepack": "pinst --disable",
     "postpack": "pinst --enable",
-    "removedb": "rm -rf .entropy && docker compose --file dev/docker-scripts/two-nodes.yaml down 2> /dev/null"
+    "publish:rc": "yarn burn && yarn build && npm publish --no-tag",
```
Comment (mixmix) [inline package.json:67]:
```suggestion
    "publish:rc": "yarn burn && yarn build && npm publish --tag next",
```

https://docs.npmjs.com/cli/v10/commands/npm-publish#tag

**Pull request merged**